istioctl/cmd/istioctl/gen_binding.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	multierror "github.com/hashicorp/go-multierror"
+	"github.com/spf13/cobra"
+
+	"istio.io/istio/istioctl/pkg/genbinding"
+	"istio.io/istio/pilot/pkg/model"
+)
+
+const (
+	defaultEgressGatewayHolder = "%default%"
+	defaultEgressGateway       = "istio-egressgateway.istio-system"
+)
+
+var (
+	remoteClusters []string
+	addressLabels  string
+	useEgress      bool
+	egressGateway  string
+
+	genBindingCmd = &cobra.Command{
+		Use:     "gen-binding",
+		Short:   "Generate Istio Configuration to direct traffic to another Istio mesh",
+		Long:    "Generate Istio Configuration to direct traffic to another Istio mesh.",
+		Example: "istioctl experimental gen-binding <service:port> --cluster <ip:port> [--cluster <ip:port>]* [--labels key1=value1,key2=value2] [--use-egress] [--egressgateway <ip:port>]",
```

Comment:
do we need a config nob for mTLS enabled or off?

Reply:
This works only with tls enabled.